x-pack/plugins/observability/public/data/slo/common.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SLOWithSummaryResponse } from '@kbn/slo-schema';
+
+export const buildOccurrencesObjective = (
+  params: Partial<SLOWithSummaryResponse['objective']> = {}
+): SLOWithSummaryResponse['objective'] => {
```

Comment:
Do I understand correctly that this is a `io-ts` type? And will typescript use it as a return type?

Reply:
SLOWithSummaryResponse is actually a `t.OutputOf<typeof some-io-ts-schema>` which becomes a proper TS `type`.